### PR TITLE
Dynamic visual (overlay addition) code

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request

 - **This is basically introducing a concept rather than requesting a pull.**

Adds Dynamic visual code
It's basically `vis_contents` thing, but it allows you to see different sprites based on your visibility level
This may be possible to used in alternative hud, but I am not sure yet, so I just posted this anyway.

### why not using `SSvis_overlays.add_vis_overlay()`?
because it needs icon, iconstate, blabla to add it into vis_contents, but this one is specifically made for adding yourself as add_overlay then to apply it to yourself again, but also different appearance based on your visibility level.

I mean, providing `(icon, iconstate, layer, plane, dir, alpha, add_appearance_flags)` is not possible, but also you need 

meanwhile, I am not sure about the performance cost.
Maybe it should be a subsystem that works different from vis_overlay subsystem...

### Additional case to add

we can add "aura" thing to stuff.
for example, something is very looks normal, but somewhere between 25(mob default)~60(spirit), we can let them to see some aura from an item as long as their visibility qualifies.

 * i.e.)
Chaplain sword looks very normal sword.
but once it accepts a spirit, it starts to emit aura (at 50 visibility level)
and people with 50 see_visible level can see if the sword has a mind by seeing the aura.

 * i.e.2)
Put Halo to chaplains
Only specific people can see their halo

 * i.e.3)
an anomaly aura from an item that you can see when you wear glasses

maybe it can be great if it's compatible well with alternative hud.

## Why It's Good For The Game
better overlay management

## Testing Photographs and Procedure
image

basically, it's invisible to everyone when you're invisible
you need to use vis_contents to let overlay is only visible to observers

image

It's what I see myself when I aghost

## Changelog
:cl:
add: dynamic visual atom for overlay thing
tweak: Now you can see stealth-agent-box, and invisible cloaked person as observer
tweak: you'll be automatically orbit-transferred to the pop-up man in a box when the box is opened
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
Hmm, maybe `alternate_appearance` should be fancier than doing this
